File: 정수론 및 조합론/2981/2981.py
import math

def gcd(a, b):
   return gcd(b, a % b) if b else a

n = int(input())
arr = [None] * n
for i in range(n):
   arr[i] = int(input())

m = 0
for i in range(n):
   for j in range(i + 1, n):
      m = gcd(abs(arr[i] - arr[j]), m)

answer = []
for i in range(1, int(math.sqrt(m)) + 1):
   if m % i == 0:
      answer.append(i)
      answer.append(m // i)
answer.sort()
for i in range(1, len(answer)):
   if answer[i] != answer[i - 1]:
      print(answer[i], end = ' ')

File: 정수론 및 조합론/2981/2981.cpp
#include <cmath>
#include <iostream>
#include <vector>
using namespace std;

int gcd(int, int);

int main()
{
   int n, m = 0;
   cin>>n;
   vector<int> v(n);
   for (int i = 0; i < n; ++i)
      cin>>v[i];

   for (int i = 0; i < n; ++i)
      for (int j = i + 1; j < n; ++j)
         m = gcd(abs(v[i] - v[j]), m);

   for (int i = 2; i * i < m; ++i)
      if (m % i == 0)
         cout<<i<<" ";
   for (int i = sqrt(m); i > 0; --i)
      if (m % i == 0)
         cout<<m / i<<" ";

   return 0;
}

int gcd(int a, int b)
{
   return b ? gcd(b, a % b) : a;
}